src/workerd/api/http.c++

```diff
@@ -1965,7 +1981,29 @@ jsg::Promise<jsg::Ref<Response>> handleHttpRedirectResponse(
   //   - Translates HEAD requests that hit 303 into HEAD requests with null bodies.
   //   - Translates all other requests that hit 303 into GET requests with null bodies.
 
-  auto redirectedLocation = urlList.back().tryParseRelative(location);
+  auto redirectedLocation = ([&]() -> kj::Maybe<kj::Url> {
+    // TODO(later): This is a bit unfortunate. Per the fetch spec, we're supposed to be
+    // using standard WHATWG URL parsing to resolve the redirect URL. However, changing it
+    // now requires a compat flag. In order to minimize changes to the rest of the impl
+    // we end up double parsing the URL here, once with the standard parser to produce the
+    // correct result, and again with kj::Url in order to produce something that works with
+    // the existing code. Fortunately the standard parser is fast but it would be nice to
+    // be able to avoid the double parse at some point.
+    if (FeatureFlags::get(js).getFetchStandardUrl()) {
+      auto base = urlList.back().toString();
```

Comment:
Should we change `urlList` to store strings rather than `kj::Url` objects?

Reply:
will make this change in a separate follow up PR